esdb/types.go

```diff
@@ -405,88 +480,87 @@ func StreamMetadataFromMap(props map[string]interface{}) (StreamMetadata, error)
 				continue
 			}
 
-			return meta, fmt.Errorf("invalid $maxCount value: %v", value)
+			return nil, fmt.Errorf("invalid $maxCount value: %v", value)
 		case "$maxAge":
 			if secs, ok := lookForUint64(value); ok {
 				meta.SetMaxAge(time.Duration(secs) * time.Second)
 				continue
 			}
 
-			return meta, fmt.Errorf("invalid $maxAge value: %v", value)
+			return nil, fmt.Errorf("invalid $maxAge value: %v", value)
 		case "$tb":
 			if i, ok := lookForUint64(value); ok {
 				meta.SetTruncateBefore(i)
 				continue
 			}
 
-			return meta, fmt.Errorf("invalid $tb value: %v", value)
+			return nil, fmt.Errorf("invalid $tb value: %v", value)
 		case "$cacheControl":
 			if secs, ok := lookForUint64(value); ok {
 				meta.SetCacheControl(time.Duration(secs) * time.Second)
 				continue
 			}
 
-			return meta, fmt.Errorf("invalid $cacheControl value: %v, type: %t", value, value)
+			return nil, fmt.Errorf("invalid $cacheControl value: %v, type: %t", value, value)
 		case "$acl":
 			switch aclValue := value.(type) {
 			case string:
 				if aclValue != UserStreamAcl && aclValue != SystemStreamAcl {
-					return meta, fmt.Errorf("invalid string $acl value: %v", aclValue)
+					return nil, fmt.Errorf("invalid string $acl value: %v", aclValue)
 				}
 
 				meta.SetAcl(value)
 			case map[string]interface{}:
-				acl, err := AclFromMap(aclValue)
+				acl, err := aclFromMap(aclValue)
 
 				if err != nil {
-					return meta, err
+					return nil, err
 				}
 
 				meta.SetAcl(acl)
 			default:
-				return meta, fmt.Errorf("invalid $acl object value: %v", value)
+				return nil, fmt.Errorf("invalid $acl object value: %v", value)
 			}
 
 		default:
 			meta.AddCustomProperty(key, value)
 		}
 	}
 
-	return meta, nil
+	return &meta, nil
 }
 
+// FilterType  represents the type filters supported by EventStoreDB.
 type FilterType int
 
 const (
-	EventFilterType   FilterType = 0
-	StreamFilterType  FilterType = 1
-	NoMaxSearchWindow int        = -1
+	// EventFilterType the filter will be applied on an event's type.
+	EventFilterType FilterType = 0
+	// StreamFilterType the filter will be applied on a stream name.
+	StreamFilterType FilterType = 1
+	// NoMaxSearchWindow disables the max search window.
+	NoMaxSearchWindow int = -1
 )
 
+// SubscriptionFilter is a filter that targets $all stream.
 type SubscriptionFilter struct {
-	Type     FilterType
+	// Type of filter.
+	Type FilterType
+	// String prefixes.
 	Prefixes []string
-	Regex    string
+	// Regex expression.
+	Regex string
 }
 
+// ExcludeSystemEventsFilter filters out event's with system's event types.
```

Comment:
Should this include the fact that system event types start with a `$`?
```suggestion
// ExcludeSystemEventsFilter filters out event's with system's event types, i.e. event types starting with `$`.
```